toto/toto-run.py

```diff
@@ -19,87 +19,125 @@
+  ./toto-run.py --step-name write-code --materials . --products . --key bob \
+      -- vi foo.py
   ```
 
-  The actual wrapper and the tasks it performs are implemented in runlib.
-
 """
 
 import os
 import sys
 import argparse
 import toto.util
 import toto.runlib
-
+import toto.log as log
+
+def _die(msg, exitcode=1):
+  log.error(msg)
+  sys.exit(exitcode)
+
+def in_toto_run(step_name, key_path, material_list, product_list,
+    link_cmd_args, record_byproducts=False):
+  """Load link signing private keys from disk and runs passed command, storing
+  its materials, by-products and return value, and products into link metadata
+  file. The link metadata file is signed and stored to disk. """
+  try:
+    log.doing("load link signing key...")
+    key = toto.util.prompt_import_rsa_key_from_file(key_path)
+  except Exception, e:
+    _die("in load key - %s" % e)
+
+  try:
+    log.doing("record materials...")
+    materials_dict = toto.runlib.record_artifacts_as_dict(material_list)
+  except Exception, e:
+    _die("in record materials - %s" % e)
+
+  try:
+    log.doing("run command...")
+    byproducts, return_value = toto.runlib.execute_link(link_cmd_args,
+        record_byproducts)
+  except Exception, e:
+    _die("in run command - %s" % e)
+
+  try:
+    log.doing("record products...")
+    products_dict = toto.runlib.record_artifacts_as_dict(product_list)
+  except Exception, e:
+    _die("in record products - %s" % e)
+
+  try:
+    log.doing("create link metadata...")
+    link = toto.runlib.create_link_metadata(step_name, materials_dict,
+        products_dict, link_cmd_args, byproducts, return_value)
+  except Exception, e:
+    raise e
+
+  try:
+    log.doing("sign metadata...")
+    link.sign(key)
+  except Exception, e:
+    _die("in sign metadata - %s" % e)
+
+  try:
+    log.doing("store metadata...")
+    link.dump()
+  except Exception, e:
+    _die("in store metadata - %s" % e)
 
 def main():
-  # Create new parser with custom usage message
   parser = argparse.ArgumentParser(
       description="Executes link command and records metadata",
-      usage="python -m %s --name <unique name>\n" \
-            "            [--materials <filepath>[,<filepath> ...]]\n" \
-            "             --products <filepath>[,<filepath> ...]\n" \
-            "             --key <filepath>\n" \
-            "             --record-byproducts\n" \
-            "             -- <cmd> [args]" % (os.path.basename(__file__), ))
-
-  # Option group for toto specific options, e.g. material and product
-  toto_args = parser.add_argument_group("Toto options")
+      usage="./toto-verify.py --step-name <unique step name>\n" \
+            "                [--materials <filepath>[,<filepath> ...]]\n" \
+            "                [--products <filepath>[,<filepath> ...]]\n" \
+            "                 --key <functionary private key path>\n" \
+            "                [--record-byproducts]\n" \
```

Comment:
Maybe it's better to enclose multi-line strings using parentheses, instead of `\`. I personally prefer it because of the reasons explained [here](https://stackoverflow.com/questions/3346230/wrap-long-lines-in-python/18160132#18160132). I don't think this is a blocker though.